assets/js/oauth.js

```diff
@@ -24,17 +30,30 @@ const sendPostRequest = async (url, params) => {
     throw err;
   }
 };
-// Parse a query string into an object
-const parseQueryString = string => {
-  if (string === "") {
+
+/**
+ * Parse a query string into an object
+ *
+ * @param {String} searchQuery - The search query params
+ * @returns {Object}
+ */
+
+const parseQueryString = searchQuery => {
+  if (searchQuery === "") {
     return {};
   }
-  const segments = string.split("&").map(s => s.split("="));
+  const segments = searchQuery.split("&").map(s => s.split("="));
   const queryString = {};
   segments.forEach(s => (queryString[s[0]] = s[1]));
```

Comment:
```js
  const queryString = segments.reduce(
    (obj, el) => ({ ...obj, [el[0]]: el[1] }),
    {}
  );
```

Reply:
@liyasthomas what do you think about the above approach?

Reply:
Does it make the code hard to read:question:

Reply:
I thought `forEach` made more sense but going through `reduce` implementation, I think devs prefer `reduce` over `forEach` to make new object. So I :+1:  that.